src/js/game/LevelMVC/LevelModel.js

```diff
@@ -35,20 +35,17 @@ module.exports = class LevelModel {
     this.shadingPlane = [];
     this.actionPlane = new LevelPlane(this.initialLevelData.actionPlane, this.planeWidth, this.planeHeight, this.controller, this, true);
 
-    for (let i = 0; i < this.actionPlane._data.length; ++i) {
-      if (this.actionPlane._data[i].blockType === "railsRedstoneTorch") {
-        let position = this.actionPlane.indexToCoordinates(i);
+    this.actionPlace.getAllPositions().forEach((position) => {
+      if (this.actionPlane.getBlockAt(position).blockType === "railsRedstoneTorch") {
         this.actionPlane.redstonePropagation(position);
       }
-    }
-    for (let i = 0; i < this.actionPlane._data.length; ++i) {
-      if (this.actionPlane._data[i].blockType.substring(0,12) === "redstoneWire") {
-        let y = Math.floor(i / this.planeHeight);
-        let x = i - (y * this.planeHeight);
```

Comment:
note that this was previously using `planeHeight` when I'm pretty sure we actually want x and y to be determined by planeWidth